### ISSUE TITLE
Support time-travel / fast-forwarding

### ISSUE THREAD

**Issue by chadoh** [near/runner#1]
Give tests an interface to jump forward N blocks

**Comment by willemneal** [comment]
@ailisp Is this currently possible with the patch state method?

**Comment by ailisp** [comment]
@willemneal no, patch state is applied per block like transactions, it cannot instruct to produce several blocks at once. The time travel.
@chadoh To be more clear, time travel and fast forwarding are two things. Time travel means to change the next block's timestamp, for example, three days after. Fast forwarding is produce N blocks at once so next block height is N more than current block. We have a mostly working time-travel implementation: https://github.com/near/nearcore/pull/3661, but some tests are not fixed to land it. We don't have a mechanism in near to fast forward. If we do want, I can add it to sandbox's requirement and implement it

**Comment by ailisp** [comment]
@chadoh I realized what described in this issue is "fast-forwarding", and fast-forwarding is easier to implement so I'll work on it first. Arbitrary time travel, if needed, I suggest address in a later issue

**Comment by willemneal** [comment]
@ailisp We just need fast-forwarding for now.  I'm definitely down to help with that once you get started!

**Comment by ailisp** [comment]
@willemneal Thanks for confirming! How soon will you need fast forwarding on nearcore side?

**Comment by willemneal** [comment]
@ailisp we just need it by the end of the quarter.

**Comment by jberrytech** [comment]
Blocker followup notes. Need core team support for this. Connecting with Bo to understand.

**Comment by TrevorJTClarke** [comment]
> it just produce one block at height N0+N and continue produce blocks as usual.

Hello, just chiming in on the need (except over on workspaces-rs). 
Producing empty blocks seems like overkill, when it would be much more efficient to jump block height as suggested. It seems this would cause a major change to nearcore to be possible.

For reference, our use case needs fast forward because all core features are time based for cron.cat, where we are testing the scheduling of transactions gets executed after its supposed to.

**Comment by ailisp** [comment]
Yes I agree jump block height is more efficient and what we should implement! And I'd say fast forward is quite fit your use case
